x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_export_csv_modal/index.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import type { EuiBasicTableColumn, EuiSearchBarProps, EuiTableSelectionType } from '@elastic/eui';
+import {
+  EuiConfirmModal,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiNotificationBadge,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { startCase, uniqBy } from 'lodash';
+
+import { AGENT_FIELDS_TO_EXPORT, INITIAL_AGENT_FIELDS_TO_EXPORT } from './columns';
+
+export interface ExportField {
+  field: string;
+}
+
+interface Props {
+  onClose: () => void;
+  onSubmit: (columns: ExportField[]) => void;
+  agentCount: number;
+}
+
+export const AgentExportCSVModal: React.FunctionComponent<Props> = ({
+  onClose,
+  onSubmit,
+  agentCount,
+}) => {
+  const [selection, setSelection] = useState<ExportField[]>(INITIAL_AGENT_FIELDS_TO_EXPORT);
+
+  const items = AGENT_FIELDS_TO_EXPORT
+    ? uniqBy([...INITIAL_AGENT_FIELDS_TO_EXPORT, ...AGENT_FIELDS_TO_EXPORT], 'field')
+    : INITIAL_AGENT_FIELDS_TO_EXPORT;
+
+  const columns: Array<EuiBasicTableColumn<ExportField>> = [
+    {
+      field: 'field',
+      name: 'Field',
+      truncateText: true,
+    },
+    {
+      field: 'field',
+      name: 'Description',
+      render: (field: string) => {
```

Comment:
nit: what do you think of having a translated description colocated with the field definition?

Reply:
thanks for the suggestion, updated